src/Stratis.SmartContracts.Core.Validation/Validators/Method/MethodParamValidator.cs

```diff
@@ -66,7 +66,13 @@ public static (bool, string) IsValidParam(MethodDefinition methodDefinition, Par
                 return (true, null);
             }
 
-            var allowedType = AllowedTypes.Contains(param.ParameterType.FullName);
+            // In private methods we allow structs
+            if (!methodDefinition.IsPublic && param.ParameterType.IsValueType)
```

Comment:
We can allow all param types here if the method is not public.

Reply:
Decided against that - weird to have a private method that allows DateTime but is hence unusable?

Reply:
It will still fail validation if a method param is a disallowed Type, so that should be picked up before deployment.

Otherwise it would be possible to write a method that uses a Type which is not an allowed public method param Type, but _is_ allowed internally (eg. int[]). One consequence of this is that methods could then not be further broken down into smaller pieces.